opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -14,124 +14,181 @@
 
 from abc import ABC, abstractmethod
 from collections import OrderedDict
+from enum import IntEnum
 from logging import getLogger
 from math import inf
+from threading import Lock
+from typing import Generic, Optional, Sequence, TypeVar
 
-from opentelemetry._metrics.instrument import _Monotonic
+from opentelemetry.sdk._metrics.measurement import Measurement
+from opentelemetry.sdk._metrics.point import Gauge, Histogram, PointT, Sum
 from opentelemetry.util._time import _time_ns
 
+
+class AggregationTemporality(IntEnum):
+    UNSPECIFIED = 0
+    DELTA = 1
+    CUMULATIVE = 2
+
+
+_PointVarT = TypeVar("_PointVarT", bound=PointT)
+
 _logger = getLogger(__name__)
 
 
-class Aggregation(ABC):
-    @property
-    def value(self):
-        return self._value  # pylint: disable=no-member
+class Aggregation(ABC, Generic[_PointVarT]):
+    def __init__(self, is_monotonic: bool):
+        self._is_monotonic = is_monotonic
```

Comment:
Can we remove `is_monotonic` from the superclass and just move it into the applicable subclasses?

Reply:
Ok, I removed it. I added a new class `_MonotonicitySensitiveAggregation` that is a parent of `XSumAggregation` and `ExplicitBucketHistogramAggregation` classes. This may look like overkill now but I think in next PRs we will face the need to identify if an aggregation is "sensitive" to the corresponding instrument monotonicity. This is because if the aggregation is sensitive, then we will need to pass an additional parameter to the constructor of the aggregation. We could instead ask if the aggregation is a sum or histogram aggregation but this is not correct, since that means asking not if the aggregation is sensitive to the instrument monotonicity but something else. If a new instrument monotonicity aggregation is added we would need to update the code in 2 places instead of one.

Reply:
> The more I think about, I also am not sure it even makes sense to have is_monotonic in the constructor. For example, if you are using a Histogram instrument to measure request byte size and reconfigure that instrument with Sum aggregator, it should output a monotonic sum.

Hm, this sounds like there can be conflicting configurations (I assume you refer to Views reconfiguring instruments). This is a very good point, actually.

I imagine this can be a lenghty topic to discuss at the spec level. For the time being, how about we log a warning and don't apply a conflicting configuration?

Reply:
What do you mean conflicting configuration? I think that example is a valid configuration.

Reply:
> For example, if you are using a `Histogram` instrument to measure request byte size and reconfigure that instrument with Sum aggregator, it should output a monotonic sum.

Ok, maybe I am not understanding your point here. Do you mean that in some circumstances (like this one) the `is_monotonic` argument becomes trivial or meaningless? What about other situations where it is still relevant?

Reply:
I see what you're saying, I didn't know Histogram instruments only allows non-negative measurements.

Reply:
How about we (in a separate issue/PR) implement detecting invalid configurations and logging the corresponding warning?